### PR TITLE
More schema documentation and cleanup

### PR DESCRIPTION
**Pull request by fabsx00**
* Sort CPG schema files
* Rename schema files for consistency
* Move more properties from base to more specific schema files
* Add `code` field to `AstNode`

### CODE REVIEW

File: schema/src/main/scala/io/shiftleft/codepropertygraph/schema/FileSystem.scala [hunk 1]
```diff
@@ -4,23 +4,23 @@ import overflowdb.schema.{NodeType, SchemaBuilder, SchemaInfo}
...
+            typeDeclSchema: Type.Schema) =
     new Schema(builder, base, namespaces, methodSchema, typeDeclSchema)
 
   class Schema(builder: SchemaBuilder,
                base: Base.Schema,
-               namespaces: Namespaces.Schema,
+               namespaces: Namespace.Schema,
                methodSchema: Method.Schema,
-               typeDeclSchema: TypeDecl.Schema) {
+               typeDeclSchema: Type.Schema) {
```
Comment (mpollmeier) [inline schema/src/main/scala/io/shiftleft/codepropertygraph/schema/FileSystem.scala:23]:
```suggestion
               typeSchema: Type.Schema) {
```

File: schema/src/main/scala/io/shiftleft/codepropertygraph/schema/FileSystem.scala [hunk 2]
```diff
@@ -4,23 +4,23 @@ import overflowdb.schema.{NodeType, SchemaBuilder, SchemaInfo}
...
     """
       |""".stripMargin
 
   def apply(builder: SchemaBuilder,
             base: Base.Schema,
-            namespaces: Namespaces.Schema,
+            namespaces: Namespace.Schema,
             methodSchema: Method.Schema,
-            typeDeclSchema: TypeDecl.Schema) =
+            typeDeclSchema: Type.Schema) =
```
Comment (mpollmeier) [inline schema/src/main/scala/io/shiftleft/codepropertygraph/schema/FileSystem.scala:16]:
```suggestion
            typeSchema: Type.Schema) =
```

File: schema/src/main/scala/io/shiftleft/codepropertygraph/schema/Method.scala [hunk 3]
```diff
@@ -5,10 +5,10 @@ import overflowdb.storage.ValueTypes
 
 object Method extends SchemaBase {
 
-  def apply(builder: SchemaBuilder, base: Base.Schema, typeDeclSchema: TypeDecl.Schema) =
+  def apply(builder: SchemaBuilder, base: Base.Schema, typeDeclSchema: Type.Schema) =
```
Comment (mpollmeier) [inline schema/src/main/scala/io/shiftleft/codepropertygraph/schema/Method.scala:8]:
```suggestion
  def apply(builder: SchemaBuilder, base: Base.Schema, typeSchema: Type.Schema) =
```

File: schema/src/main/scala/io/shiftleft/codepropertygraph/schema/Method.scala [hunk 4]
```diff
@@ -17,7 +17,7 @@ object Method extends SchemaBase {
       | This layer is provided by the frontend and may be modified by passes.
       |""".stripMargin
 
-  class Schema(builder: SchemaBuilder, base: Base.Schema, typeDeclSchema: TypeDecl.Schema) {
+  class Schema(builder: SchemaBuilder, base: Base.Schema, typeDeclSchema: Type.Schema) {
```
Comment (mpollmeier) [inline schema/src/main/scala/io/shiftleft/codepropertygraph/schema/Method.scala:20]:
```suggestion
  class Schema(builder: SchemaBuilder, base: Base.Schema, typeSchema: Type.Schema) {
```

**Review by mpollmeier — APPROVED** [review]
a few more cases of s/typeDeclSchema/typeDecl/

File: schema/src/main/scala/io/shiftleft/codepropertygraph/schema/Ast.scala [hunk 5]
```diff
@@ -102,12 +102,22 @@ object Ast extends SchemaBase {
     val literal: NodeType = builder
       .addNodeType(
         name = "LITERAL",
-        comment = "Literal/Constant"
+        comment = "Literal/Constant. This may be a string, in which case it includes the "
```
[the remark this answers is not in the document]
Comment (fabsx00) [reply schema/src/main/scala/io/shiftleft/codepropertygraph/schema/Ast.scala]: Fixed

**Pull request merged**